### PR TITLE
redesign the repo structure for the better professional environment

### PR DESCRIPTION
**Pull request by Mentoriassssssss**
I have redesign the repository structure. The purpose is to make the repository more detailed, structured and organized.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
-    <div className="App">
-      <p>Project has initialized successfully. <br/> Now do the work you lazy fuckers.</p> 
-    </div>
+      <Router>
+        <div className="App">
+          <Routes>
+            {publicRoutes.map((route, index) => {
+              const Layout = route.layout || defaultLayout;
+              const Page = route.component;
+              console.log(Page)
```
Comment (kietpt230) [inline src/App.tsx:13]: Remove console.log, please

**Review by kietpt230 — COMMENTED** [review]
I see some files are empty. Can you explain it for me, please ?

File: src/App.tsx (hunk 1 above)
Comment (kietpt230) [inline src/App.tsx]: Remove console.log, please

File: src/components/Header/header.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const Header : React.FC<HeaderProps> = () => {
+    return(
+        <div className="header_container">
+            Header
+        </div>
+    )
+}
+
+export default Header;
```
Comment (kietpt230) [inline src/components/Header/header.tsx:14]: Add end of line, please

File: src/components/Sidebar/sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+const Sidebar : React.FC<SidebarProps> = () => {
+    return(
+        <div className="sidebar_container">
+            Sidebar
+        </div>
+    )
+}
+
+export default Sidebar;
```
Comment (kietpt230) [inline src/components/Sidebar/sidebar.tsx:14]: Add end of line, please

File: src/layouts/defaultLayout/index.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+            <div className="deflay_main">
+                <Sidebar/>
+                <div className="deflay_main_content">{children}</div>
+            </div>
+        </div>
+    )
+
+}
+
+export default defaultLayout;
```
Comment (kietpt230) [inline src/layouts/defaultLayout/index.tsx:18]: Add end of line, please

File: src/pages/following.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+const Home: React.FC<HomeProps> = () => {
+    return (
+        <div className="home_container">
+            Home
+        </div>
+    )
+}
+
+export default Home;
```
Comment (kietpt230) [inline src/pages/following.tsx:13]: Add end of line, please

File: src/pages/home.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import '../css/pages/home/home.css'
+
+const Home = () => {
+    return (
+    <div>
+        Home
+    </div>)
+}
+
+export default Home;
```
Comment (kietpt230) [inline src/pages/home.tsx]: Add end of line, please

File: src/routes/index.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+const publicRoutes : publicRoute[] = [
+    {path:'/', component: Home},
+    {path:'/following', component: Following},
+]
+
+const privateRoutes : privateRoute[] = [
+]
+
+export {publicRoutes, privateRoutes}
```
Comment (kietpt230) [inline src/routes/index.tsx:24]: Add end of line, please

**Review by kietpt230 — CHANGES_REQUESTED** [review]

**Comment by kietpt230** [comment]
image
Please, set the label correspond to the status of the PR

**Comment by darkplayer0211** [comment]
Hi @whbmc152,
cc @kietpt230,

Please support the idea of abstraction as discussed in the last Weekly Scrum.

You can add the **ProcessCore** folder in **components** and import it in **App.tsx**.

Or, you can separate the UI in the foreground with the **ui_src** folder and the processing in the background with the **process_src** folder.

Besides, please update the **README.md** file for the new folder structure too.

Best regards,
Tri Nguyen

**Review by darkplayer0211 — CHANGES_REQUESTED** [review]
Hi @whbmc152,
cc @kietpt230,

Please support the idea of abstraction as discussed in the last Weekly Scrum.

You can add the **ProcessCore** folder in **components** and import it in **App.tsx**.

Or, you can separate the UI in the foreground with the **ui_src** folder and the processing in the background with the **process_src** folder.

Besides, please update the **README.md** file for the new folder structure.

Best regards,
Tri Nguyen

**Pull request merged**